conftest.py

```diff
@@ -401,3 +410,24 @@ def _upload_cluster_dump(client, cluster):
                 client.dump_cluster_state(dump_path, **s3_storage_options)
 
     yield _upload_cluster_dump
+
+
+@pytest.fixture
+def upload_performance_report(test_run_benchmark):
+    @contextlib.contextmanager
+    def _upload_performance_report(name):
+        with coiled.performance_report(name) as obj:
+            yield
+        # TODO: This will not scale well if we have many reports. Instead, the
+        # performance report ctx manager should somehow yield an object that has
+        # the url
+        if obj is None:
```

Comment:
I opened https://gitlab.com/coiled/cloud/-/merge_requests/4985 to make this more accessible.

Reply:
I agree that this should be exposed via the `coiled` API.

I wonder if this would be easier today if we capture stdout from the context manager cleanup, which includes the URL

Reply:
if the upstream PR is accepted I'd rather keep it as is until this is deployed. I doubt that we'll accumulate a significant amount of reports until this version is rolled out and I'd rather not mess with stdout parsing

Reply:
FWI upstream was merged